src/main/java/life/qbic/expdesign/io/QBiCDesignReader.java

```diff
@@ -25,15 +27,15 @@
 
 public class QBiCDesignReader implements IExperimentalDesignReader {
 
-  private ArrayList<String> vocabulary = new ArrayList<String>(Arrays.asList("Identifier", "SPACE",
-      "EXPERIMENT", "SAMPLE TYPE", "Q_SECONDARY_NAME", "PARENT"));
-  private List<String> entityMandatory = new ArrayList<String>(Arrays.asList("Q_NCBI_ORGANISM"));
-  private List<String> extractMandatory = new ArrayList<String>(Arrays.asList("Q_PRIMARY_TISSUE"));
-  private List<String> extractSpecials = new ArrayList<String>(Arrays.asList("Q_TISSUE_DETAILED"));
-  private List<String> testMandatory = new ArrayList<String>(Arrays.asList("Q_SAMPLE_TYPE"));
-  private List<String> mhcSpecials = new ArrayList<String>(Arrays.asList("Q_MHC_CLASS"));
+  private ArrayList<String> vocabulary = new ArrayList<>(Arrays.asList("Identifier", "SPACE",
+      "EXPERIMENT", "SAMPLE TYPE", OpenbisPropertyCodes.Q_SECONDARY_NAME.name(), "PARENT"));
+  private List<String> entityMandatory = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_NCBI_ORGANISM.name()));
+  private List<String> extractMandatory = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_PRIMARY_TISSUE.name()));
+  private List<String> extractSpecials = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_TISSUE_DETAILED.name()));
+  private List<String> testMandatory = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_SAMPLE_TYPE.name()));
+  private List<String> mhcSpecials = new ArrayList<>(Arrays.asList(OpenbisPropertyCodes.Q_MHC_CLASS.name()));
   private List<String> sampleTypesInOrder =
-      new ArrayList<String>(Arrays.asList("Q_BIOLOGICAL_ENTITY", "Q_BIOLOGICAL_SAMPLE",
+      new ArrayList<>(Arrays.asList("Q_BIOLOGICAL_ENTITY", "Q_BIOLOGICAL_SAMPLE",
           "Q_TEST_SAMPLE", "Q_MHC_LIGAND_EXTRACT", "Q_NGS_SINGLE_SAMPLE_RUN", "Q_MS_RUN"));
 
```

Comment:
Can the content here (the codes) also be replaced?

Reply:
indeed, good idea